File: invision-backend/app/Enums/CampaignStatus.php
<?php

namespace App\Enums;

enum CampaignStatus: string
{
    case Draft = 'draft';
    case Scheduled = 'scheduled';
    case Active = 'active';
    case Paused = 'paused';
    case Completed = 'completed';
    case Cancelled = 'cancelled';

    public function label(): string
    {
        return match ($this) {
            self::Draft => 'Draft',
            self::Scheduled => 'Scheduled',
            self::Active => 'Active',
            self::Paused => 'Paused',
            self::Completed => 'Completed',
            self::Cancelled => 'Cancelled',
        };
    }

    public function color(): string
    {
        return match ($this) {
            self::Draft => 'gray',
            self::Scheduled => 'blue',
            self::Active => 'green',
            self::Paused => 'yellow',
            self::Completed => 'indigo',
            self::Cancelled => 'red',
        };
    }
}

File: invision-mobile/lib/src/core/enums/campaign_status.dart
enum CampaignStatus {
  draft,
  scheduled,
  active,
  paused,
  completed,
  cancelled;

  String get label => switch (this) {
    draft => 'Draft',
    scheduled => 'Scheduled',
    active => 'Active',
    paused => 'Paused',
    completed => 'Completed',
    cancelled => 'Cancelled',
  };

  static CampaignStatus fromString(String value) {
    return switch (value) {
      'draft' => CampaignStatus.draft,
      'scheduled' => CampaignStatus.scheduled,
      'active' => CampaignStatus.active,
      'paused' => CampaignStatus.paused,
      'completed' => CampaignStatus.completed,
      'cancelled' => CampaignStatus.cancelled,
      _ => CampaignStatus.draft,
    };
  }
}
